src/main/java/com/marginallyclever/makelangelo/makeArt/io/vector/LoadScratch3.java

```diff
@@ -104,207 +161,384 @@ public Turtle load(InputStream in) throws Exception {
+		}
+		
+		myStack.push(list);
+	}
+
+	private Scratch3Procedure findProcedureWithProccode(String proccode) {
+		Iterator<Scratch3Procedure> iter = scratchProcedures.iterator();
+		while(iter.hasNext()) {
+			Scratch3Procedure p = iter.next();
+			if(p.proccode.equals(proccode)) return p;
+		}
+		return null;
+	}
+
+	private void doIf(JSONObject currentBlock) throws Exception {
+		logger.debug("IF");
+		String condition = (String)findInputInBlock(currentBlock,"CONDITION");
+		String substack = (String)findInputInBlock(currentBlock,"SUBSTACK");
+		if(resolveBoolean(getBlock(condition))) {
+			parseScratchCode(substack);
+		}
+	}
+
+	private void doRepeatForever(JSONObject currentBlock) throws Exception {
+		throw new Exception(Translator.get("LoadScratch3.foreverNotAllowed"));
+		// technically this would work and the program would never end.  It is here for reference.
+		//logger.debug("REPEAT FOREVER");
+		//String substack = (String)findInputInBlock(currentBlock,"SUBSTACK");
+		//while(true) {
+		//	parseScratchCode(substack);
+		//}
+	}
+
+	private void doRepeatUntil(JSONObject currentBlock) throws Exception {
+		logger.debug("REPEAT UNTIL");
+		String condition = (String)findInputInBlock(currentBlock,"CONDITION");
+		String substack = (String)findInputInBlock(currentBlock,"SUBSTACK");
+		
+		while(!resolveBoolean(getBlock(condition))) {
+			parseScratchCode(substack);
+		}
+	}
+
+	private void doRepeat(JSONObject currentBlock) throws Exception {
+		int count = (int)resolveValue(findInputInBlock(currentBlock,"TIMES"));
+		String substack = (String)findInputInBlock(currentBlock,"SUBSTACK");
+		logger.debug("REPEAT {}",count);
+		for(int i=0;i<count;++i) {
+			parseScratchCode(substack);
+		}		
+	}
+
+	// relative change
+	private void changeVariableBy(JSONObject currentBlock) throws Exception {
+		Scratch3Variable v = getScratchVariable((String)findFieldsInBlock(currentBlock,"VARIABLE"));
+		double newValue = resolveValue(findInputInBlock(currentBlock,"VALUE"));
+		// set and report
+		v.value = (double)v.value + newValue;
+		logger.debug("Set {} to {}", v.name, v.value);
+	}
+
+	// absolute change
+	private void setVariableTo(JSONObject currentBlock) throws Exception {
+		Scratch3Variable v = getScratchVariable((String)findFieldsInBlock(currentBlock,"VARIABLE"));
+		double newValue = resolveValue(findInputInBlock(currentBlock,"VALUE"));
+		// set and report
+		v.value = newValue;
+		logger.debug("Set {} to {}", v.name, v.value);
+	}
+
+	private void doMotionGotoXY(JSONObject currentBlock) throws Exception {
+		double px = resolveValue(findInputInBlock(currentBlock,"X"));
+		double py = resolveValue(findInputInBlock(currentBlock,"Y"));
+		logger.debug("GOTO {} {}",px,py);
+		myTurtle.moveTo(px, py);
+	}
+
+	private void doMotionPointInDirection(JSONObject currentBlock) throws Exception {
+		double v = resolveValue(findInputInBlock(currentBlock,"DIRECTION"));
+		logger.debug("POINT AT {}",v);
+		myTurtle.setAngle(v);
```

Comment:
`myTurtle.setAngle(v-90.0);//a orientation 90° diffrence between Turtle and Scratch`
seem to resolv my orientation issue.